test/unit/api.test.js

```diff
@@ -0,0 +1,203 @@
+'use strict';
+
+const sinon = require('sinon');
+const { expect } = require('chai');
+
+const mongodbDriver = require('mongodb');
+const mongodbLegacy = require('../..');
+const { MongoDBNamespace } = require('mongodb/lib/utils');
+const { asyncApi } = require('../tools/api');
+
+// A map of class names to their list of method details
+const classesToMethods = new Map(
+  asyncApi.map((api, _, array) => [
```

Comment:
so this is just one array of methods, what about the ones listed under transformMethods?
and there are a bunch of methods with the special note that it transforms something, but they aren't repeated under transformMethods, and not everything in the transformMethods is in asynApi

I'm just really confused about the structure of which APIs we're testing and how we're testing those other ones that aren't in the list

probably should add a description in tools that explains exactly what each of those two exported arrays is supposed to be representing

Reply:
I have restructured the contents of api.js to be just the one array as our source of truth for the APIs to test, and added more descriptive flags to the ones that can't be automatically tested. 

now the calculated exports are just the classNames and a map of className to methods